engine/apps/api/views/user.py

```diff
@@ -113,49 +104,53 @@ class UserView(
         PluginAuthentication,
     )
 
-    permission_classes = (IsAuthenticated, ActionPermission)
+    permission_classes = (IsAuthenticated, RBACPermission)
 
     # Non-admin users are allowed to list and retrieve users
     # The overridden get_serializer_class will return
     # another Serializer for non-admin users with sensitive information hidden
-    action_permissions = {
-        IsAdminOrEditor: (
-            *MODIFY_ACTIONS,
-            "list",
-            "metadata",
-            "verify_number",
-            "forget_number",
-            "get_verification_code",
-            "get_backend_verification_code",
-            "get_telegram_verification_code",
-            "unlink_slack",
-            "unlink_telegram",
-            "unlink_backend",
-            "make_test_call",
-            "export_token",
-            "mobile_app_verification_token",
-            "mobile_app_auth_token",
-        ),
-        AnyRole: ("retrieve", "timezone_options"),
+    rbac_permissions = {
+        # TODO: what permissions should go here?
+        "list": [],
+        "retrieve": [],
+        "create": [],
+        "update": [RBACPermission.Permissions.OWN_SETTINGS_WRITE],
```

Comment:
create and update/partial_update should be `RBACPermission.Permissions.OWN_SETTINGS_WRITE`